File: tools/package.py
#!/usr/bin/env python3
"""Empaquette skill/ en dist/lingo-english-tutor.zip, prêt à importer dans l'app Claude.

Le ZIP doit contenir le dossier du skill à sa racine, et ce dossier doit porter le nom
déclaré dans le frontmatter — pas 'skill'. C'est pourquoi le dossier est renommé ici
plutôt que dans le dépôt : garder 'skill/' en local évite de confondre le dossier source
avec le paquet installable.

Usage :  python3 tools/package.py
"""
import os, re, sys, zipfile, yaml

SRC, DIST = "skill", "dist"

raw = open(f"{SRC}/SKILL.md", encoding="utf-8").read()
m = re.match(r"^---\n(.*?)\n---\n", raw, re.S)
if not m:
    sys.exit("SKILL.md : frontmatter absent")
name = yaml.safe_load(m.group(1))["name"]

os.makedirs(DIST, exist_ok=True)
out = f"{DIST}/{name}.zip"

with zipfile.ZipFile(out, "w", zipfile.ZIP_DEFLATED) as z:
    for f in sorted(os.listdir(SRC)):
        path = os.path.join(SRC, f)
        if os.path.isfile(path):
            z.write(path, arcname=f"{name}/{f}")          # <- dossier nommé à la racine

with zipfile.ZipFile(out) as z:
    entries = z.namelist()

print(f"{out}  ({os.path.getsize(out) / 1024:.0f} Ko)")
for e in entries:
    print("   ", e)

roots = {e.split("/")[0] for e in entries}
if roots != {name}:
    sys.exit(f"ERREUR : racine du ZIP = {roots}, attendu {{{name}}}")
if f"{name}/SKILL.md" not in entries:
    sys.exit("ERREUR : SKILL.md absent de la racine du dossier")
print("\nStructure conforme.")

File: tools/package.mjs
#!/usr/bin/env node
/**
 * Packages `skill/` into `dist/<skill-name>.zip`, ready to import into the
 * Claude app and into Cowork.
 *
 *   npm run zip
 *
 * The source directory is named `skill/`, but the archive must contain a
 * directory named after the frontmatter's declared name. The rename happens
 * here rather than in the repo: keeping `skill/` locally avoids confusing the
 * source directory with the installable package.
 */
import { mkdirSync, writeFileSync } from 'node:fs';
import { createZip, readZipNames } from '../src/zip.mjs';
import { skillName, zipEntries } from '../src/skill.mjs';

const name = skillName();
const out = `dist/${name}.zip`;

const archive = createZip(zipEntries());
mkdirSync('dist', { recursive: true });
writeFileSync(out, archive);

const entries = readZipNames(archive);
console.log(`${out}  (${Math.round(archive.length / 1024)} Ko)`);
for (const e of entries) console.log('   ', e);

const problems = [];
const roots = new Set(entries.map((e) => e.split('/')[0]));
if (roots.size !== 1 || !roots.has(name)) {
  problems.push(`racine du ZIP = {${[...roots].join(', ')}}, attendu {${name}}`);
}
if (!entries.includes(`${name}/SKILL.md`)) {
  problems.push('SKILL.md absent de la racine du dossier');
}

if (problems.length) {
  console.error();
  for (const p of problems) console.error('ERREUR :', p);
  process.exit(1);
}
console.log('\nStructure conforme.');
